src/qz/utils/ConnectionUtilities.java

```diff
@@ -54,6 +65,36 @@ public static InputStream getInputStream(String urlString) throws IOException {
         }
     }
 
+    private static boolean isAllowed(String allowed, URL url) {
+        if(url == null) return false;
+        String urlProtocol = url.getProtocol();
+        if(urlProtocol == null || urlProtocol.trim().isEmpty()) return false;
+        allowed = ArgValue.SECURITY_DATA_PROTOCOLS.getDefaultVal() +
+                (allowed == null || allowed.trim().isEmpty() ? "" : "," + allowed);
+        String[] protocols = allowed.split(",");
+        // Loop over http, https, etc
+        for(String protocol : protocols) {
+            if (url != null) {
```

Comment:
probably dont need this null check here since this method wouldve already returned false if it was null